staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -848,6 +853,8 @@ func (c *Cacher) dispatchEvent(event *watchCacheEvent) {
 	if event.Type == watch.Bookmark {
 		for _, watcher := range c.watchersBuffer {
 			watcher.nonblockingAdd(event)
+			// Requeue the watcher for the next bookmark.
+			c.bookmarkWatchers.addWatcher(watcher)
```

Comment:
Wouldn't be cleaner to do it in line 924 and update the comment there?

Reply:
done